lib/Conversion/TorchToTcp/Utils.cpp

```diff
@@ -76,3 +76,161 @@ Value torch_to_tcp::broadcastInLeadingDimsToMatchShape(
 
   return result;
 }
+
+// Example: [] -> [N, C, H, W]
+Value torch_to_tcp::broadcast0DToNDAndMatchShape(
+    ConversionPatternRewriter &rewriter, Value input, Value target) {
+
+  RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+  RankedTensorType targetType = target.getType().cast<RankedTensorType>();
+
+  // This utility only accepts 0D inputs
+  assert(inputType.getRank() == 0);
+
+  Value result = input;
+
+  // First: Broadcast Rank
+  // [] -> [1, 1, 1, 1]
+  // reassociation map = [[]]
+  SmallVector<ReassociationExprs> reassociationMap(inputType.getRank());
+  SmallVector<int64_t> resultShape(targetType.getRank(), 1);
+  auto resultType = targetType.cloneWith(makeArrayRef(resultShape),
+                                         targetType.getElementType());
+  result = rewriter.create<tensor::ExpandShapeOp>(
+      result.getDefiningOp()->getLoc(), resultType, input, reassociationMap);
+  // Second: Broadcast Shape
+  // [1, 1, 1, 1] -> [N, C, H, W]
+  SmallVector<int64_t> axes;
+  SmallVector<Value> dimSizes;
+  for (int64_t axis = 0; axis < targetType.getRank(); ++axis) {
+    axes.push_back(axis);
+    dimSizes.push_back(rewriter.createOrFold<tensor::DimOp>(
+        result.getDefiningOp()->getLoc(), target, axis));
+  }
+  auto axesAttr = rewriter.getI64ArrayAttr(axes);
+  result = rewriter.create<tcp::BroadcastOp>(result.getDefiningOp()->getLoc(),
+                                             target.getType(), result, dimSizes,
+                                             axesAttr);
+
+  return result;
+}
+
+// Example: [C] -> [N, C, H, W]
+Value torch_to_tcp::broadcast1DToNDAndMatchShape(
```

Comment:
The implementation of the 1D and 0D case are almost identical. All that is needed is two `if` statements conditioned on the input rank to create the reassociations and the `dimSizes`. I would refactor all the logic into a helper function.

Reply:
Done, PTAL.